### ISSUE TITLE
E0623 not working for below examples

### ISSUE THREAD

**Issue by gaurikholkar** [rust-lang/rust#43875]
Example 1
```
struct Foo;

impl Foo {
    fn foo(&self, v: &mut Vec<&u32>, x: &u32) {
        v.push(x);
    }
}
```

Example 2
```
fn foo(v: &mut Vec<&u32>, x: &u32) {
    v.push(x);
}
```

Example 3
```
struct Foo;

impl Foo {
    fn foo(&self, x: &u32) -> &u32 {
        x
    }
}
```
cc @nikomatsakis

**Comment by gaurikholkar** [comment]
first two work now, third one will be fixed in an ongoing PR.

**Comment by gaurikholkar** [comment]
Done

**Comment by arielb1** [comment]
Merged in #44936.

**Comment by estebank** [comment]
I believe the current output is reasonable for all the presented cases.
